src/js/components/PlacementConstraintsSchemaField.js

```diff
@@ -83,6 +110,20 @@ export default class PlacementConstraintsSchemaField extends Component {
     const { label, errorMessage } = this.props;
     const { batch } = this.state;
 
+    const identity = (memo, transaction) => memo.concat(transaction);
+    const hasBatchErrors = !PlacementValidators.validateNoBatchError(
+      batch.reduce(identity, [])
+    );
+
+    if (hasBatchErrors) {
+      const fieldName = this.props.fieldProps.name;
+      const fieldValue = this.props.fieldProps.formData;
```

Comment:
💯

Reply:
I noticed that behavior too. I think when the parenthesis is not balanced this failure happens on an earlier stage and we receive an empty/null information that does not even come to the parser. 

I will see if I can narrow down this case quickly and understand why it happens.

Reply:
I checked this by adding some debugging to the JSON parsers. They do not get triggered when the string is not JSON. That tells me that the step that verifies that is up in the flow, and hence I would have to implement extra logic there to catch this case.

I am creating a ticket to improve on this.
https://jira.mesosphere.com/browse/DCOS-20624

Reply:
Thanks for creating ticket. I agree, not something for this PR 👍